src/app/util/process-global-message.cpp

```diff
@@ -154,203 +151,8 @@ bool emAfProcessGlobalCommand(EmberAfClusterCommand * cmd)
-        appResponseLength = (cmd->mfgSpecific ? 4 : 2);
-        FALLTHROUGH;
-        /* fall through */
-    // DO NOT BREAK from this case
-
-    // the format of the write attributes cmd is:
-    // ([attr ID:4] [data type:1] [data:N]) * N
-    // the format of the write attributes response is:
-    // ([status 1] [attr ID 2]) * n
-    // ONLY errors are reported unless all are successful then a single success
-    // is sent. write attr no response is handled by just executing the same
-    // code but not setting the flag that sends the response at the end.
-    case ZCL_WRITE_ATTRIBUTES_NO_RESPONSE_COMMAND_ID:
-    case ZCL_WRITE_ATTRIBUTES_COMMAND_ID: {
-        uint8_t numFailures = 0;
-        uint8_t numSuccess  = 0;
-        uint8_t dataType;
-        uint16_t dataSize;
-#if (BIGENDIAN_CPU)
-        uint8_t writeData[ATTRIBUTE_LARGEST];
-#endif //(BIGENDIAN_CPU)
-        EmberAfStatus status;
-
-        // set the cmd byte - this is byte 3 index 2, but since we have
-        // already incremented past the 3 byte ZCL header (our index is at 3),
-        // this gets written to "-1" since 3 - 1 = 2.
-        emberAfPutInt32uInResp(ZCL_WRITE_ATTRIBUTES_RESPONSE_COMMAND_ID);
-
-        // go through the message until there are no more attrID/type/data
-        while (msgLen > msgIndex + 5)
-        {
-            attrId   = emberAfGetInt32u(message, msgIndex, msgLen);
-            dataType = emberAfGetInt8u(message, msgIndex + 4, msgLen);
-
-            dataSize = emberAfAttributeValueSize(clusterId, attrId, dataType, message + msgIndex + 5);
-
-            // the data is sent little endian over-the-air, it needs to be
-            // inserted into the table big endian for the EM250 and little
-            // endian for the EZSP hosts. This means for the EM250 the data
-            // needs to be reversed before sending to writeAttributes
-#if (BIGENDIAN_CPU)
-            if (dataSize <= msgLen - (msgIndex + 5) && dataSize <= ATTRIBUTE_LARGEST)
-            {
-                // strings go over the air as length byte and then in human
-                // readable format. These should not be flipped.
-                if (emberAfIsThisDataTypeAStringType(dataType))
-                {
-                    memmove(writeData, message + msgIndex + 5, dataSize);
-                }
-                else
-                {
-                    // the data is sent little endian over-the-air, it needs to be
-                    // inserted into the table big endian
-                    uint16_t i;
-                    for (i = 0; i < dataSize; i++)
-                    {
-                        writeData[i] = message[msgIndex + 5 + dataSize - i - 1];
-                    }
-                }
-#else  //(BIGENDIAN_CPU)
-            if (dataSize <= msgLen - (msgIndex + 5))
-            {
-#endif //(BIGENDIAN_CPU)
-
-                status = emberAfWriteAttributeExternal(cmd->apsFrame->destinationEndpoint, clusterId, attrId, clientServerMask,
```

Comment:
That is going to break some use cases that relies on external storage.

Reply:
@tlykkeberg-grundfos for introducing this code.

Reply:
After some reading, seems this function does some kind of ACL checks.
In this case, I think https://github.com/project-chip/connectedhomeip/issues/8442 should update related code and add tests to support ACL function aligned with spec.
On the other hand, I updated `WriteSingleClusterData` to use `emberAfWriteAttributeExternal` instead of `emberAfWriteAttribute`.